crates/compute/src/cpu/layer.rs

```diff
@@ -7,14 +7,77 @@ use binius_field::{
 };
 
 use super::{memory::CpuMemory, tower_macro::each_tower_subfield};
-use crate::layer::{ComputeLayer, Error, FSlice, FSliceMut};
+use crate::{
+	layer::{ComputeLayer, Error, FSlice, FSliceMut},
+	memory::FSliceWithTowerLevel,
+};
 
 #[derive(Debug)]
 pub struct CpuExecutor;
 
 #[derive(Debug, Default)]
 pub struct CpuLayer<F: TowerFamily>(PhantomData<F>);
 
+impl<F: TowerFamily> CpuLayer<F> {
+	pub fn new() -> Self {
+		Self(PhantomData)
+	}
+}
+
+/// evals is treated as a matrix with `1 << log_query_size` rows and each column is dot-producted
+/// with the corresponding query element. The result is written to the `output` slice of packed values.
+fn compute_left_fold<EvalType: TowerField, T: TowerFamily>(
+	evals_as_b128: Vec<T::B128>,
+	log_evals_size: usize,
+	query: Vec<T::B128>,
+	log_query_size: usize,
+	out: FSliceMut<'_, T::B128, CpuLayer<T>>,
+) -> Result<(), Error>
+where
+	<T as TowerFamily>::B128: From<EvalType> + ExtensionField<EvalType>,
+{
+	let evals = evals_as_b128
+		.iter()
+		.flat_map(<T::B128 as ExtensionField<EvalType>>::iter_bases)
+		.collect::<Vec<_>>();
```

Comment:
In case of 1b eval elements, that would collect a `Vec` of `BinaryField1b` elements, which are definitely not `Pod`.